### DISCUSSION TITLE
blog/javascript/6-closure

### DISCUSSION THREAD

**Discussion by giscus** [pySoo/Next-blog#80] [General]
# blog/javascript/6-closure

JavaScript의 클로저의 원리를 쉽게 배워보고 어떤 상황에 사용되는지 알아보도록 하겠습니다.

https://enjoydev.life/blog/javascript/6-closure

**Comment by magrowing** [comment]
클로저에 대해 공부하면서 항상 어려웠는데 글을 통해 쉽게 이해 할 수 있었습니다! 좋은 글 감사합니다!

**Reply by pySoo** [reply]
안녕하세요! 이해하시는데 도움이 됐다니 정말 기쁘네요 ㅎㅎ
댓글도 너무 감사합니다 ☺️

**Comment by Im-younique** [comment]
안녕하세요, 좋은 글 잘 읽고 갑니다! 쉽게 잘 설명해 주신 것 같아서 많은 도움 받고갑니다. 🔥🔥🔥

**Reply by pySoo** [reply]
안녕하세요 동윤님! 도움이 되셨다니 다행입니다 🔥
댓글 감사드립니다 ☺️

**Comment by te-ing** [comment]
작성된지 오래된 글이지만 질문남깁니다..!
리액트에서 렌더링 될 때 함수가 재호출되면서 사라진 이전 상태값을 클로저를 통해 가져올 수 있다고 하셨는데, 
제 생각에는 리액트에서 상태값이 변경될 때, 이에 따른 가상돔을 생성하고 반영하는 것이기 때문에 이전 상태값은 새로운 상태가 렌더링 될 때 까지 사라지지 않는다고 생각해요.
혹시 이전 상태값을 클로저를 통해 가져온다고 말씀하신 이유가 있을까요?
